### DISCUSSION TITLE
load stm32duino as components/library in stmcube

### DISCUSSION THREAD

**Discussion by tsctrl** [stm32duino/Arduino_Core_STM32#1499] [Ideas]
i come from esp where i am able to use arduino as components in espressif ide to speed up the development work and at the same time have flexibility to access low level api as required.

i might not look enough but i cant find any step to load stm32duino as library in stm cube in the description.

do stm32duino support this? any guide for me to get started? 

thanks!

**Comment by fpistm** [comment]
Hi @tsctrl 
Currently not. 
I've already thought about this and maybe it will come. 🤞 
Time is always the missing ingredient.

**Reply by tsctrl** [reply]
Thanks @fpistm. The idea is to reuse Arduino codebase/library without having to work on different platform. yes the .ino might or might not work in cube but at least we have access to all the library possibility.

**Comment by ABOSTM** [comment]
` ... at the same time have flexibility to access low level api as required.`
This is possible within Arduino IDE: you can use stm32cube HAL/LL directly from your Arduino Sketch

**Reply by tsctrl** [reply]
noted and thanks @ABOSTM. is it fully supported? including hardware encryption, security and also tweaking default pin configuration such changing pin functionality?
my concern is on the critical api that was not accessible in Arduino IDE. 

Sorry, I have limited knowledge on STM and have not go thru all capabilities in Arduino IDE vs Cube.

**Reply by ABOSTM** [reply]
`Fully supported,` yes, but as far as HAL is concerned, you may need to enable the concerned Peripheral: https://github.com/stm32duino/wiki/wiki/HAL-configuration
`changing pin configuration` you can customize your  PeripheralPins : https://github.com/stm32duino/wiki/wiki/Custom-definitions#custom-pinmap-array

`I have limited knowledge on STM`   It is advised to have some knowledge to use stm32cube HAL/LL which are more complex than Arduino API

**Comment by fpistm** [comment]
Here you can find an example of an STM32CubeFw example converted to Arduino:
https://www.stm32duino.com/viewtopic.php?f=41&t=110

**Comment by fpistm** [comment]
With last version of the STM32CubeIDE, CMake is supported. Since the STM32 core also deployed CMake, it is now possible to load it to CubeIDE.
